grails-app/controllers/org/broadinstitute/orsp/api/ProjectController.groovy

```diff
@@ -40,8 +41,17 @@ class ProjectController extends AuthenticatedController {
         String projectKey = params.id
         Gson gson = new Gson()
         Object input = gson.fromJson(gson.toJson(request.JSON), Object.class)
-        Issue updatedIssue = issueService.modifyExtraProperties(input, projectKey)
-        render([message: updatedIssue] as JSON)
+        try {
+            Issue updatedIssue = issueService.modifyExtraProperties(input, projectKey)
+            if (input.get(IssueExtraProperty.PROJECT_REVIEW_APPROVED)) {
```

Comment:
I've tried to keep all property getters in Issue - I think this could be there.
In Issue (change to a boolean if that's what this is):
```
transient String getProjectReviewApproved() { getExtraProperties().find { it.name == IssueExtraProperty.PROJECT_REVIEW_APPROVED }?.value }
```
Then here:
```suggestion
            if (issue.getProjectReviewApproved()) {
```

Reply:
Makes sense to have a "Project Review Approved" getter in Issue, I added as you suggested. Eitherway the idea here was to be able to call `modifyExtraProperties` with a generic input object. Since `input` can have any extra property we only call `updateProjectApproval` if the current input is intended to change this state.

If we receive an input without "PROJECT_REVIEW_APPROVED" constant, but the issue/project (`updatedIssue` in this case) already has this property, it will verify again all approvals, which won't break anything... anyhow will take some extra db calls.

Please check again the change I made. This check is now performed in `modifyExtraProperties` as you can see [here](https://github.com/broadinstitute/orsp-pub/blob/BTRX-528-generalApprove/grails-app/services/org/broadinstitute/orsp/IssueService.groovy#L204)